lib/utils/manager/download_manager/download_manager.dart

```diff
@@ -0,0 +1,232 @@
+
+typedef FutureVoidCallback = Future<void> Function();
+
+class DownloadManager {
+  DownloadManager._();
+
+  static final DownloadManager _instance = DownloadManager._();
+
+  factory DownloadManager() => _instance;
+
+  final workingQueue = getIt.get<DownloadWorkerQueue>();
+
+  final mediaAPI = getIt.get<MediaAPI>();
+
+  final Map<String, DownloadFileInfo> _eventIdMapDownloadFileInfo = {};
+
+  void cancelDownload(String eventId) async {
+    final cancelToken = _eventIdMapDownloadFileInfo[eventId]?.cancelToken;
+    if (cancelToken != null) {
+      cancelToken.cancel();
+      _eventIdMapDownloadFileInfo[eventId]?.downloadStateStreamController.add(
+            const Right(
+              DownloadFileInitial(),
+            ),
+          );
+      _clear(eventId);
+    }
+  }
+
+  void _initDownloadFileInfo(
+    Event event,
+  ) {
+    final streamController = StreamController<Either<Failure, Success>>();
+
+    _eventIdMapDownloadFileInfo[event.eventId] = DownloadFileInfo(
+      eventId: event.eventId,
+      cancelToken: CancelToken(),
+      downloadStateStreamController: streamController,
+      downloadStream: streamController.stream.asBroadcastStream(),
+    );
+  }
+
+  Stream<Either<Failure, Success>>? getDownloadStateStream(String eventId) {
+    return _eventIdMapDownloadFileInfo[eventId]?.downloadStream;
+  }
+
+  Future<void> _clear(String eventId) async {
+    try {
+      await _eventIdMapDownloadFileInfo[eventId]
+          ?.downloadStateStreamController
+          .close();
+    } catch (e) {
+      Logs().e(
+        'DownloadManager::_clear(): $e',
+      );
+      rethrow;
+    } finally {
+      _eventIdMapDownloadFileInfo.remove(eventId);
+    }
+  }
+
+  Future<void> download({
+    required String savePath,
+    required Event event,
+    bool getThumbnail = false,
+  }) async {
+    _initDownloadFileInfo(event);
+    final streamController = _eventIdMapDownloadFileInfo[event.eventId]
+        ?.downloadStateStreamController;
+    final cancelToken = _eventIdMapDownloadFileInfo[event.eventId]?.cancelToken;
+    if (streamController == null || cancelToken == null) {
+      Logs().e(
+        'DownloadManager::download(): streamController or cancelToken is null',
+      );
+      return;
+    }
+    streamController.add(
+      const Right(
+        DownloadFileInitial(),
+      ),
+    );
+    if (!PlatformInfos.isWeb) {
+      workingQueue.addTask(
+        Task(
+          id: event.eventId,
+          runnable: () async {
+            try {
+              await event.getFileInfo(
+                getThumbnail: getThumbnail,
+                progressCallback: (receive, total) async {
+                  if (total != 0 && receive == total) {
+                    await _handleDownloadFileDone(
+                      event,
+                      savePath,
+                      streamController,
+                      receive,
+                      total,
+                    );
```

Comment:
still not have cleanup when completed task

Reply:
clean in message_download_content now, after the view have receive the DownloadSuccessState